src/GitHub/Data/Traffic.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE KindSignatures    #-}
+{-# LANGUAGE LambdaCase        #-}
+
+-- | Data types used in the traffic API
+module GitHub.Data.Traffic where
+
+import Data.Text   (Text)
+import Data.Time   (UTCTime)
+import Data.Vector (Vector)
+
+import GitHub.Data.Name        (Name)
+import GitHub.Internal.Prelude
+import Prelude ()
+
+data Referrer = Referrer
+    { referrer        :: !(Name Referrer)
+    , referrerCount   :: !Int
+    , referrerUniques :: !Int
+    }
+    deriving (Eq, Show, Generic)
+
+instance FromJSON Referrer where
+    parseJSON = withObject "Referrer" $ \o ->
+      Referrer
+      <$> o .: "referrer"
+      <*> o .: "count"
+      <*> o .: "uniques"
+
+data PopularPath = PopularPath
+    { popularPath        :: !Text
+    , popularPathTitle   :: !Text
+    , popularPathCount   :: !Int
+    , popularPathUniques :: !Int
+    }
+    deriving (Eq, Show)
+
+instance FromJSON PopularPath where
+    parseJSON = withObject "Path" $ \o ->
+      PopularPath
+      <$> o .: "path"
+      <*> o .: "title"
+      <*> o .: "count"
+      <*> o .: "uniques"
+
+data Period =
+    Day
+    | Week
+    deriving (Eq, Show)
+
+prettyPeriod :: IsString a => Period -> a
```

Comment:
Looks like that previously we didn't define such functions, or if did that they are unexported internals of `Endpoints` module.

Please move it there.

Let's not tie pretty-printing with serialising to API-value. (accidentally the same).